### DISCUSSION TITLE
TOC fully expanded

### DISCUSSION THREAD

**Discussion by stefansimik** [quarto-dev/quarto-cli#3726] [Q&A]
Is it possible to have TOC fully expanded and see all levels?
Now TOC automatically expands and collapses, based on where we click.

I am asking in general, but specially interested in HTML export format.

**Comment by mcanouil** [comment]
If you are looking for a `toc-collapse` option to be set to false, I am afraid, there isn't right now.

@dragonstyle Do you think this option could be added? (it exists in `rmarkdown`)
I looked at the codebase and if I am not mistaken, currently it is hardcoded.
https://github.com/quarto-dev/quarto-cli/blob/aeb98c47089177f6e25ef7e23a95ca4d6854350a/src/format/html/format-html-bootstrap.ts#L306-L320

**Reply by stefansimik** [reply]
That is exactly, what I am looking for 👍
It would be great and very useful, if this could be implemented 👌 .

**Reply by aborruso** [reply]
I think it's important to have it, great idea!

**Comment by fernandovmacedo** [comment]
I was wondering about this same question, and It would be great to have this option. While this doesn't happen, I have crafted a hack to make it work in the HTML format.

I have created a `custom.scss` file with the following rule:
```
/*-- scss:rules --*/
.sidebar nav[role="doc-toc"] ul {
  display: block;
}
```
And added after the default theme:
```
format: 
  html:
    toc: true
    theme:
      - default
      - custom.scss
```
With this, the TOC will always be expanded. I'm using that until the option become available.

**Reply by dragonstyle** [reply]
The next pre-release will support `toc-depth-expanded` which will allow you specific what level to expand in HTML table of contents. It defaults to 1.

**Reply by mcanouil** [reply]
does this support an option to expand "all"?

**Reply by dragonstyle** [reply]
If you set it to your max depth (or higher) it will :)

**Reply by mcanouil** [reply]
The idea was to avoid setting a "weird" value such as 99 😅
For example, true would be 99 or whatever.

**Reply by dragonstyle** [reply]
Maybe I could rename it to `toc-expanded`which would make more sense like:

```
toc-expanded: true
toc-expanded: 3
```

What do you think?

**Reply by dragonstyle** [reply]
(the above should now be supported in the next pre-release build)

**Reply by mcanouil** [reply]
Maybe `toc-expansion` to follow the syntax of other `toc-*` keys.
With: 
> Specify the number of section levels to be expanded in the table of contents. The default is X. 
>
> Note that if set to `true`, all section levels will be expanded in the table of contents.

**Reply by dragonstyle** [reply]
I ended up with `toc-expand: true` or `toc-expand: false` or `toc-expand: 3` :)
